### ISSUE TITLE
docker image with "full-scan" don't obey to -m parameter

### ISSUE THREAD

**Issue by ramarro123** [zaproxy/zaproxy#8694]
### Describe the bug:

while running docker image, latest, in various host machine, the scan doesn't come to end even if using the -m switch that should limit the scan time

what i got are some connection error, at http host (but i passed https, that's expected? i doubt my host reply correctly to http or it should redirect to https)

the error, repeated forever (till i kill the docker run)

`org.apache.hc.client5.http.impl.classic.ZapHttpRequestRetryExec - Recoverable I/O exception (org.apache.hc.core5.http.NoHttpResponseException) caught when processing request to {}->http://www.fabrick.com:80`

### Steps to reproduce the behavior:

docker run -v $(pwd):/zap/wrk/:rw    -t ghcr.io/zaproxy/zaproxy:stable zap-full-scan.py -t https://www.fabrick.com -r zap.html -m 1

### Expected behavior:

obey to -m switch, and even in case of error, end the scan with the expected time, and eventually report the error somehow.

also again is strange that it try http instead of https, but maybe is expected

### Software versions:

ghcr.io/zaproxy/zaproxy:stable

### Would you like to help fix this issue?

- [ ] Yes

**Comment by ramarro123** [comment]
let me add another bit, fabrick.com to http call reply with

gbs05515@mac0288 ~ % curl http://www.fabrick.com -vv
* Host www.fabrick.com:80 was resolved.
* IPv6: (none)
* IPv4: 13.107.253.72
*   Trying 13.107.253.72:80...
* Connected to www.fabrick.com (13.107.253.72) port 80
> GET / HTTP/1.1
> Host: www.fabrick.com
> User-Agent: curl/8.7.1
> Accept: */*
>
* Request completely sent off
< HTTP/1.1 307 Temporary Redirect
< Date: Thu, 31 Oct 2024 08:46:42 GMT
< Content-Type: text/html
< Content-Length: 0
< Connection: keep-alive
< Location: https://www.fabrick.com/
< x-azure-ref: 20241031T084642Z-166dfbc85496lg56wzhbdaaf1c0000000b0g000000002cyn
< X-Cache: CONFIG_NOCACHE

not sure if it help

**Comment by psiinon** [comment]
Are there any errors in the zap.log file?
https://www.zaproxy.org/docs/docker/diagnosing-problems/#zap-fails-for-another-reason

**Comment by ramarro123** [comment]
not that i can see, it's extremely tricky to get them.

but with -m 5 shouldn't the job exit after 5 min or i am understanding it wrongly?


if i run
$ /zap/zap-full-scan.py -t https://www.fabrick.com/ -r zap.html -m 5 || true
WARNING: step_script could not run to completion because the timeout was exceeded. For more control over job and script timeouts see: https://docs.gitlab.com/ee/ci/runners/configure_runners.html#set-script-and-after_script-timeouts
ERROR: Job failed: execution took longer than 1h0m0s seconds


after 1hr the runner die for timeout

i wonder if that's expected, maybe i am understanding wrongly how it should work.

**Comment by kingthorin** [comment]
It isn't forced to stop, it just means that no further work will be started. If zap happens to be analyzing a huge response at the time the limit is hit then it can appear thing or to not obey the limit.

**Comment by psiinon** [comment]
Thats true, but taking an hour to exit with a 5 min limit of active scanning is surprising, and feels wrong..

**Comment by ramarro123** [comment]
i actually try to increase the limit, to 2hr but the situation don't change.

there is a way to put it in "verbose" mode to get some realtime log that could be useful?

connecting to the docker image isn't trivial when it run from the gitlab docker env on the runner unfortunatley

thanks!

**Comment by thc202** [comment]
There's the `-d` flag.

**Comment by ramarro123** [comment]
the log is quite useless at least to me, here an extract

it's running for 2hr

here a piece of log
2024-11-05 11:38:43,876 Params: ['/zap/zap-x.sh', '-daemon', '-port', '36594', '-host', '0.0.0.0', '-config', 'database.recoverylog=false', '-config', 'api.disablekey=true', '-config', 'api.addrs.addr.name=.*', '-config', 'api.addrs.addr.regex=true', '-config', 'spider.maxDuration=5', '-addonupdate', '-addoninstall', 'pscanrulesBeta', '-addoninstall', 'ascanrulesBeta']


2024-11-05 11:41:28,269 http://localhost:36594 "GET http://zap/JSON/ascan/view/status/?scanId=0 HTTP/1.1" 200 14
2024-11-05 11:41:28,270 Active Scan progress %: 2
2024-11-05 11:41:33,278 Starting new HTTP connection (1): localhost:36594
.....
2024-11-05 11:56:11,432 http://localhost:36594 "GET http://zap/JSON/ascan/view/status/?scanId=0 HTTP/1.1" 200 15
2024-11-05 11:56:11,433 Active Scan progress %: 25
2024-11-05 13:35:52,723 http://localhost:36594 "GET http://zap/JSON/ascan/view/status/?scanId=0 HTTP/1.1" 200 15
2024-11-05 13:35:52,823 Active Scan progress %: 25



note that 25% seems the limit i could reach, as in 1h:30 min it's not increasing

(sorry for my bad english)
